packages/markers/src/browser/problem/problem-manager.spec.ts

```diff
@@ -5,19 +5,29 @@
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
 
+import { Container } from 'inversify';
 import * as chai from 'chai';
 import { ProblemManager } from './problem-marker';
 import URI from "@theia/core/lib/common/uri";
 import { LocalStorageService, StorageService } from '@theia/core/lib/browser/storage-service';
-import { TestLogger } from '@theia/core/lib/common/test/test-logger';
+import { ILogger } from '@theia/core/lib/common/logger';
+import { MockLogger } from '@theia/core/lib/common/test/mock-logger';
+import { FileSystemWatcher } from '@theia/filesystem/lib/common';
 
 const expect = chai.expect;
 let manager: ProblemManager;
-let store: StorageService;
+let testContainer: Container;
 
 before(async () => {
-    store = new LocalStorageService(new TestLogger());
-    manager = new ProblemManager(store, undefined);
+    testContainer = new Container();
+    testContainer.bind(ILogger).to(MockLogger);
+    testContainer.bind(StorageService).to(LocalStorageService).inSingletonScope();
+    testContainer.bind(LocalStorageService).toSelf().inSingletonScope();
+    // tslint:disable-next-line:no-any
+    testContainer.bind(FileSystemWatcher).toConstantValue(<any>undefined);
+    testContainer.bind(ProblemManager).toSelf();
+
+    manager = testContainer.get(ProblemManager);
```

Comment:
why do you want to use DI for such small contexts? It makes the code less readable and explicit.

Reply:
I guess it's subjective I think this is more consistent and readable. I'd rather always use the same method and you can see how it simplifies the widget manager test.

Reply:
This is also more explicit for example compare:

```typescript
manager = new ProblemManager(store, undefined);
```
with
```typescript
testContainer.bind(FileSystemWatcher).toConstantValue(<any>undefined);
```
In the later version you know what dependency is undefined, while in the former you would have to check the constuctor.

This also makes it independant of a constuctor refactoring you would not have to change the order of the elements in the constuctor if they were to change. It also avoids possible erros due to such a refactoring.